client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/thirdpartyclasses/NettyClassReplacement.java

```diff
@@ -0,0 +1,32 @@
+package org.evomaster.client.java.instrumentation.coverage.methodreplacement.thirdpartyclasses;
+
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.Replacement;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.ThirdPartyMethodReplacementClass;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.UsageFilter;
+import org.evomaster.client.java.instrumentation.shared.ReplacementType;
+
+
+public class NettyClassReplacement extends ThirdPartyMethodReplacementClass {
+
+    private static final NettyClassReplacement singleton = new NettyClassReplacement();
+
+    @Override
+    protected String getNameOfThirdPartyTargetClass() {
+        return "io.netty.handler.codec.http.HttpVersion";
+    }
+
+    @Replacement(replacingStatic = false,
+        type = ReplacementType.TRACKER,
+        id = "isKeepAliveDefault_boolean_class",
+        usageFilter = UsageFilter.ANY)
+    public boolean isKeepAliveDefault() {
+        /*
+            Micronaut 1.3.4 closes the connection when there is a server error
+            (http status code > 299) or isKeepAliveDefault() set to false (default
```

Comment:
clarification about this comment: server errors are codes >=500. 3xx are redirections, 4xx are user errors

Reply:
in the older version (1.3.4) under `RoutingInBoundHandler` if the `expectKeepAlive` (will be true when `isKeepAliveDefault` true or it's set to `keep-alive` during the request) boolean is `false` or HTTP status code set to greater than 299 server will send an `close` header. I've modified the comment.